tensorboard/webapp/metrics/views/card_renderer/utils.ts

```diff
@@ -96,45 +96,27 @@ export function maybeClipLinkedTimeSelection(
   minStep: number,
   maxStep: number
 ): TimeSelectionView {
+  const startStep = Math.min(
+    Math.max(minStep, timeSelection.start.step),
+    maxStep
+  );
+  const endStep = timeSelection.end
+    ? Math.max(Math.min(maxStep, timeSelection.end.step), minStep)
+    : null;
```

Comment:
Would it be worth adding comments on how do you set startStep and endStep? It does take me a while to check the logic with max wrapped in min and verse vice .